src/core/backbone_router/ndproxy_table.hpp

```diff
@@ -56,6 +57,71 @@ namespace BackboneRouter {
 class NdProxyTable : public InstanceLocator, private NonCopyable
 {
 public:
+    /**
+     * This class represents a ND Proxy instance.
+     *
+     */
+    class NdProxy : private Clearable<NdProxy>
+    {
+        friend class NdProxyTable;
+
+    public:
+        /**
+         * This method gets the Mesh-Local IID of the ND Proxy.
+         *
+         * @returns  The Mesh-Local IID.
+         *
+         */
+        const Ip6::InterfaceIdentifier &GetMeshLocalIid(void) const { return mMeshLocalIid; }
+
+        /**
+         * This method gets the time since last transaction of the ND Proxy.
+         *
+         * @returns  The time since last transaction in seconds.
+         *
+         */
+        uint32_t GetTimeSinceLastTransaction(void) const
+        {
+            return TimeMilli::MsecToSec(TimerMilli::GetNow() - mLastRegistrationTime);
+        }
+
+        /**
+         * This method gets the RLOC16 of the ND Proxy.
+         *
+         * @returns  The RLOC16 value.
+         *
+         */
+        uint16_t GetRloc16(void) const { return mRloc16; }
+
+        /**
+         * This method gets the DAD flag of the ND Proxy.
+         *
+         * @returns  The DAD flag.
+         *
+         */
+        bool GetDadFlag(void) const { return mDadFlag; }
+
+    private:
+        NdProxy(void) { Clear(); }
+
+        void Init(const Ip6::InterfaceIdentifier &aAddressIid,
+                  const Ip6::InterfaceIdentifier &aMeshLocalIid,
+                  uint16_t                        aRloc16,
+                  uint32_t                        aTimeSinceLastTransaction);
+
+        void Update(uint16_t aRloc16, uint32_t aTimeSinceLastTransaction);
+        void IncreaseDadAttampts(void) { mDadAttempts++; }
+        bool IsDadAttamptsComplete() const { return mDadAttempts == Mle::kDuaDadRepeats; }
+
+        Ip6::InterfaceIdentifier mAddressIid;
+        Ip6::InterfaceIdentifier mMeshLocalIid;
+        TimeMilli                mLastRegistrationTime; ///< in milliseconds
+        uint16_t                 mRloc16;
+        uint8_t                  mDadAttempts : 2;
+        bool                     mDadFlag : 1;
+        bool                     mValid : 1;
+    };
```

Comment:
How about adding a static-assert here to confirm that max attempts fits in 2 bits (just in case it may get changed in future - mainly since it's defined in another header file).
```
static_assert(Mle::kDuaDadRepeats < 4, "Mle::kDuaDadRepeats does not fit in mDataAttempts defined as 2-bit value");

Reply:
Actually we can already catch such issue: changing kDuaDadRepeats to 4 causes compile issues. 
```
/usr/local/google/home/simonlin/src/openthread/src/core/backbone_router/ndproxy_table.cpp: In static member function ‘static void ot::BackboneRouter::NdProxyTable::NotifyDadComplete(ot::BackboneRouter::NdProxyTable::NdProxy&, bool)’:
/usr/local/google/home/simonlin/src/openthread/src/core/backbone_router/ndproxy_table.cpp:273:38: error: conversion from ‘uint8_t’ {aka ‘unsigned char’} to ‘unsigned char:2’ changes value from ‘4’ to ‘0’ [-Werror=overflow]
  273 |         aNdProxy.mDadAttempts = Mle::kDuaDadRepeats;
      |                                      ^~~~~~~~~~~~~~
```

Reply:
static_assert added